src/core/scalebar/qgsdoubleboxscalebarrenderer.cpp

```diff
@@ -190,3 +191,29 @@ void QgsDoubleBoxScaleBarRenderer::draw( QgsRenderContext &context, const QgsSca
   //draw labels using the default method
   drawDefaultLabels( context, settings, scaleContext );
 }
+
+bool QgsDoubleBoxScaleBarRenderer::applyDefaultSettings( QgsScaleBarSettings &settings ) const
+{
+  QgsSimpleFillSymbolLayer *fill = qgis::down_cast< QgsSimpleFillSymbolLayer * >( settings.fillSymbol()->symbolLayers().at( 0 ) );
```

Comment:
qgis::down_cast will assert if the layer isn't a simple fill, so eg if the user changed to a gradient fill then this will crash:

```suggestion
  QgsSimpleFillSymbolLayer *fill = dynamic_cast< QgsSimpleFillSymbolLayer * >( settings.fillSymbol()->symbolLayers().at( 0 ) );
```